ambry-store/src/main/java/com.github.ambry.store/StatsManager.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.store;
+
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.config.StatsManagerConfig;
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Utils;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.io.JsonEncoder;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The stats manager is responsible for periodic node level aggregation of store stats and expose/publish such stats to
+ * potential consumers.
+ */
+public class StatsManager {
```

Comment:
I made it public because I placed it in the store package (just like StorageManager). It needs to be public because AmbryServer (whose initializing it) is in the server package.